xdsl/dialects/pdl_interp.py

```diff
@@ -273,6 +276,27 @@ def __init__(
             result_types=[result_type],
         )
 
+    @classmethod
+    def parse(cls, parser: Parser) -> GetResultsOp:
+        index = parser.parse_optional_integer()
+        if index is not None:
+            index = IntegerAttr.from_int_and_width(index, 32)
+        parser.parse_characters("of")
+        input_op = parser.parse_operand()
+        parser.parse_punctuation(":")
+        result_type = parser.parse_type()
+        assert isa(result_type, ValueType) or isa(result_type, RangeType[ValueType])
+        return GetResultsOp(index, input_op, result_type)
```

Comment:
This is the wrong place to do type checking, we expect this to happen during verification. I'd recommend something like this:
```suggestion
        return GetResultsOp.build(operands=(input_op,), properties={"index": index}, result_types=(result_type,))
```

Reply:
Another approach would be to raise a parsing error
```suggestion
        pos = parser.pos
	result_type = parser.parse_type()
	if not isa(result_type, ValueType | RangeType[ValueType]):
	    parser.raise_error(f"Expected value type or range of value types, got {result_type}", at_position: pos)
        return GetResultsOp(index, input_op, result_type)
```

Reply:
Sorry, took a while. I opted to not add the parsing errors.